q2_feature_table/_normalize.py

```diff
@@ -5,17 +5,26 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
-
+import random
 import biom
 
 
-def rarefy(table: biom.Table, sampling_depth: int,
-           with_replacement: bool = False) -> biom.Table:
+def rarefy(table: biom.Table,
+           sampling_depth: int,
+           with_replacement: bool = False,
+           random_seed: float = None
+           ) -> biom.Table:
+    # Generate a random seed if seed is None
+    if random_seed is not None:
+        random.seed(random_seed)
+
     if with_replacement:
         table = table.filter(lambda v, i, m: v.sum() >= sampling_depth,
                              inplace=False, axis='sample')
+
     table = table.subsample(sampling_depth, axis='sample', by_id=False,
-                            with_replacement=with_replacement)
+                            with_replacement=with_replacement,
+                            seed=random.randint(0, 2**32 - 1))
```

Comment:
I think you don't want this, as you're overriding whatever the user provided here. Also, you will have already set the seed earlier in this function. 

```suggestion
                            with_replacement=with_replacement)
```